src/Rules/Symfony/TwigTemplateExistsRule.php

```diff
@@ -0,0 +1,104 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Symfony;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Symfony\TwigEnvironmentResolver;
+use PHPStan\Type\ObjectType;
+use function count;
+use function in_array;
+use function is_string;
+use function sprintf;
+
+/**
+ * @implements Rule<MethodCall>
+ */
+final class TwigTemplateExistsRule implements Rule
+{
+
+	/** @var TwigEnvironmentResolver */
+	private $twigEnvironmentResolver;
+
+	public function __construct(TwigEnvironmentResolver $twigEnvironmentResolver)
+	{
+		$this->twigEnvironmentResolver = $twigEnvironmentResolver;
+	}
+
+	public function getNodeType(): string
+	{
+		return MethodCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$templateArg = $this->getTwigTemplateArg($node, $scope);
+
+		if ($templateArg === null) {
+			return [];
+		}
+
+		$templateNames = [];
+
+		if ($templateArg->value instanceof Variable && is_string($templateArg->value->name)) {
+			$varType = $scope->getVariableType($templateArg->value->name);
+
+			foreach ($varType->getConstantStrings() as $constantString) {
+				$templateNames[] = $constantString->getValue();
+			}
+		} elseif ($templateArg->value instanceof String_) {
+			$templateNames[] = $templateArg->value->value;
+		}
+
+		if (count($templateNames) === 0) {
+			return [];
+		}
+
+		$errors = [];
+
+		foreach ($templateNames as $templateName) {
+			if ($this->twigEnvironmentResolver->templateExists($templateName)) {
+				continue;
+			}
+
+			$errors[] = RuleErrorBuilder::message(sprintf(
+				'Twig template "%s" does not exist.',
+				$templateName
+			))->line($templateArg->getStartLine())->identifier('twig.templateNotFound')->build();
+		}
```

Comment:
I dunno if the error strategy need to depends on the PHPStan Level.

For instance, maybe `existingTemplate|nonExistingTemplate` should only be reported at level 7 or more.
In the same way unions/maybe are reported on level 7 for classic rules.